es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java

```diff
@@ -499,19 +498,21 @@ public String get(String workflowInstanceId, String fieldToGet) {
 	@Override
 	public List<String> searchArchivableWorkflows(String indexName, long archiveTtlDays) {
 		QueryBuilder q = QueryBuilders.boolQuery()
-				.must(QueryBuilders.rangeQuery("endTime").lt(LocalDate.now().minusDays(archiveTtlDays)))
 				.should(QueryBuilders.termQuery("status", "COMPLETED"))
 				.should(QueryBuilders.termQuery("status", "FAILED"))
 				.mustNot(QueryBuilders.existsQuery("archived"))
 				.minimumNumberShouldMatch(1);
 		SearchRequestBuilder s = elasticSearchClient.prepareSearch(indexName)
 				.setTypes("workflow")
 				.setQuery(q)
-				.setSize(1000);
+				.addSort("endTime", SortOrder.ASC)
+				.setSize(archiveSearchBatchSize);
 
 		SearchResponse response = s.execute().actionGet();
 
 		SearchHits hits = response.getHits();
+		logger.info("Archive search totalHits - {}", hits.getTotalHits());
```

Comment:
would it be helpful to have a metric for this one? To be able to catch it if the process returns no data for long time?